src/MeshKernelNET/Api/DisposableGriddedSamples.cs

```diff
@@ -126,8 +116,7 @@ protected override void SetNativeObject(ref GriddedSamplesNative nativeObject)
             nativeObject.coordinates_x = GetPinnedObjectPointer(coordinatesX);
             nativeObject.coordinates_y = GetPinnedObjectPointer(coordinatesY);
             nativeObject.values = GetPinnedObjectPointer(values);
-            nativeObject.upper_right_x = upperRightX;
-            nativeObject.upper_right_y = upperRightY;
+            nativeObject.values = GetPinnedObjectPointer(values);
```

Comment:
copy-paste error? duplicate line, and valueType is not set! If this wasn't detected by unit tests also a unit test is missing.

Reply:
this is the native struct   

 struct GriddedSamples
    {
        /// @brief Number of x gridded samples coordinates
        int num_x = 0;

        /// @brief Number of y gridded samples coordinates
        int num_y = 0;

        /// @brief X coordinate of the grid origin (lower left corner)
        double x_origin = 0.0;

        /// @brief Y coordinate of the grid origin (lower left corner)
        double y_origin = 0.0;

        /// @brief Constant grid cell size
        double cell_size = 0.0;

        /// @brief If not nullptr, coordinates for non-uniform grid spacing in x direction
        double* x_coordinates = nullptr;

        /// @brief If not nullptr, coordinates for non-uniform grid spacing in y direction
        double* y_coordinates = nullptr;

        /// @brief Sample values
        void* values = nullptr;

        /// @brief the numeric representation of the values (0 = short, 1 = float)
        int value_type = 1;
    };